code/utils/basic_modeling.py

```diff
@@ -2,46 +2,82 @@
 import numpy.linalg as npl
 import matplotlib.pyplot as plt
 import nibabel as nib
+from scipy.stats import t as t_dist
 
-def load_data(fname):
+def load_data(f1, f2):
     """
     Return the image data as an array and the convolved time course
 
     Parameters
     ----------
-    fname : string
+    f1,f2 : string
         The name of data to process
-
     Returns
     -------
-    tuple:
+    tuple:  
         Contains the image data as an array and the convolved time course
-
-
     """
     # Load the image as an image object
-    img = nib.load(fname + '.nii')
+    img = nib.load('../../data/sub001/BOLD/' + f1 + '.nii.gz')
     # Load the image data as an array
     # Drop the first 4 3D volumes from the array
     data = img.get_data()[..., 4:]
     # Load the pre-written convolved time course
-    convolved = np.loadtxt(fname + '_conv.txt')[4:]
-
+    convolved = np.loadtxt('../../data/convo/' + f2 + '.txt')[4:]
     return(data, convolved)
 
 def reg_voxels_4d(data, convolved):
     design = np.ones((len(convolved), 2))
     design[:, 1] = convolved
-    data_2d = np.reshape(data, (-1, data.shape[-1]))
-    betas = npl.pinv(design).dot(data_2d.T)
-    betas_4d = np.reshape(betas.T, img.shape[:-1] + (-1,))
+    data_2d = np.reshape(data, (data.shape[-1], -1))
+    betas = npl.pinv(design).dot(data_2d) #(2, 147456)
+    betas_4d = np.reshape(betas.T, data.shape[:-1] + (-1,)) #(64, 64, 36, 2)
+    print(design.shape, data_2d.shape, betas.shape)
+    return (design, data_2d, betas, betas_4d)   
+
+def RSE(X,Y, betas_hat): #input 2D of X, Y and betas_hat
+    Y_hat = X.dot(betas_hat)
+    res = Y - Y_hat
+    RSS = np.sum(res ** 2, 0)
+    print(RSS.shape)
+    df = X.shape[0] - npl.matrix_rank(X)
+    MRSS = RSS / df
 
```

Comment:
I've checked code from here above, they are all correct, the dimensions are right, and the outputs are reasonable.